### ISSUE TITLE
git-publish tries to send vim `.swp` files as patches

### ISSUE THREAD

**Issue by ehabkost** [stefanha/git-publish#27]
How to reproduce:

Step 1:

```
$ g publish -v --to ehabkost@gmail.com -b master
git rev-parse --show-toplevel
git rev-parse --git-dir
git symbolic-ref --short HEAD
git tag -l tmp/testing-gipublish-v[0-9]*
git config --get-all branch.tmp/testing-gipublish.gitpublishcc
git config branch.tmp/testing-gipublish.gitpublishcccmd
git show --raw --no-color --pretty=medium tmp/testing-gipublish-staging
git config format.coverLetter
git log --no-color --oneline master..
git config core.hooksPath
git rev-parse --git-common-dir
git show --raw --no-color --pretty=medium tmp/testing-gipublish-staging
git tag -l tmp/testing-gipublish-v[0-9]*
git show --raw --no-color --pretty=medium tmp/testing-gipublish-v0
git tag -a -m  -f tmp/testing-gipublish-staging
git config branch.tmp/testing-gipublish.gitpublishprefix
git config format.subjectprefix
git show --raw --no-color --pretty=medium tmp/testing-gipublish-staging
git log --no-color --oneline master..
git format-patch --subject-prefix PATCH --output-directory /tmp/tmpwditwe --no-cover-letter master..
Stopping so you can inspect the patch emails:
  cd /tmp/tmpwditwe

git send-email --to ehabkost@gmail.com --dry-run --confirm=never /tmp/tmpwditwe
[PATCH] TEST PATCH

To: ehabkost@gmail.com

[c] Edit Cc list in editor (save after edit)
[t] Edit To list in editor (save after edit)
[e] Edit patches in editor
[p] Print final email headers (dry run)
[a] Send all
[q] Cancel (quit)
```

Step 2: In another terminal:

```
$ cd /tmp/tmpwditwe
$ vim *
```
Step 3: Don't make any changes, but keep vim open

Step 4:

```
a
git config core.hooksPath
fatal: /tmp/tmpwditwe/.0001-TEST-PATCH.patch.swp: 1: patch contains a line longer than 998 characters
warning: no patches were sent
$ 
```

**Comment by stefanha** [comment]
Thanks for reporting this.  I'll post a fix.

**Comment by stefanha** [comment]
@ehabkost Please try this branch:
https://github.com/stefanha/git-publish/commits/fix-git-send-email-tmp-files

**Comment by ehabkost** [comment]
@stefanha it works, but are we sure users will never add/remove/rename files from the temporary dir? I think it's unlikely, but maybe if that happens we should let the user know something is wrong instead of silently ignoring new files. e.g. something like:

```diff
diff --git a/git-publish b/git-publish
index e15bbba..65c4068 100755
--- a/git-publish
+++ b/git-publish
@@ -715,6 +715,12 @@ gitpublishprofile.%s.remote''' % (topic, topic, options.profile_name)
                              options.in_reply_to, topic, options.override_cc)
 
             invoke_hook('pre-publish-send-email', tmpdir)
+            found_files = set(os.path.join(tmpdir, f) for f in os.listdir(tmpdir) if not f.startswith('.'))
+            if found_files != set(patches):
+                print "The list of files at %s changed and I don't know what to do" % (tmpdir)
+                print 'Expected: %s' % (' '.join(sorted(patches)))
+                print 'Found: %s' % (' '.join(sorted(found_files)))
+                return 1
             git_send_email(to, cc, patches, suppress_cc, options.in_reply_to)
         except (GitError, GitSendEmailError, GitHookError, InspectEmailsError):
             return 1
```

**Comment by stefanha** [comment]
glob('*') ignores dotfiles.  Is that on purpose?

The alternative would be os.listdir(tmpdir) to list all files.

**Comment by ehabkost** [comment]
Yes, it's on purpose: ignoring dotfiles (and only dotfiles) looks like a safe and useful feature. Ignoring other files doesn't seem to be safe.

**Comment by stefanha** [comment]
I have tested the following commit:
https://github.com/stefanha/git-publish/commit/92cbeeacf88339b020e1680ced7576e87bbdc795

If you're happy with this solution I'll merge the branch into master.

**Comment by stefanha** [comment]
Fixes merged (see Commit 92cbeea).
